### PR TITLE
feat(UI-1138): update auth0 connection setup

### PR DESCRIPTION
**Pull request by RonenMars**
## Description

This is a replacement of the current "default" OAuth connection, which is not possible in Auth0. In other words, this is not a new connection type.

Based on https://github.com/autokitteh/autokitteh/pull/912:

- Connection type: "OAuth" as before
- New text fields (all are required):
- ---Client ID
- ---Client Secret
- ---Auth0 Domain
- The "Start OAuth Flow" button should remain the same
- Help link:
- ---Label: "AutoKitteh guide: configuring Auth0 connections"
- ---URL: https://docs.autokitteh.com/integrations/auth0/config/

## Linear Ticket
https://linear.app/autokitteh/issue/UI-1138/update-connection-ui-for-auth0

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/connections/integrations/auth0/add.tsx [hunk 1]
```diff
@@ -2,11 +2,13 @@ import React, { useEffect } from "react";
 
 import { useTranslation } from "react-i18next";
 
-import { Integrations } from "@src/enums/components";
 import { useConnectionForm } from "@src/hooks";
-import { oauthSchema } from "@validations";
+import { auth0IntegrationSchema } from "@validations/connection.schema";
```
Comment (J1za) [inline src/components/organisms/connections/integrations/auth0/add.tsx:6]:
```suggestion
import { auth0IntegrationSchema } from "@validations";
```

File: src/components/organisms/connections/integrations/auth0/add.tsx [hunk 2]
```diff
@@ -17,23 +19,74 @@ export const Auth0IntegrationAddForm = ({
...
-	const { handleOAuth } = useConnectionForm(oauthSchema, "create");
+	const { errors, handleCustomOauth, handleSubmit, isLoading, register } = useConnectionForm(
+		auth0IntegrationSchema,
+		"create"
+	);
 
 	useEffect(() => {
 		if (connectionId) {
-			handleOAuth(connectionId, Integrations.auth0);
+			handleCustomOauth(connectionId, "auth0");
```
Comment (J1za) [inline src/components/organisms/connections/integrations/auth0/add.tsx:29]: Why not use` Integrations.auth0` ?
```suggestion
			handleCustomOauth(connectionId, Integrations.auth0);
```

**Comment by J1za** [comment]
Does the Auth0 edit form work well?

File: src/components/organisms/connections/integrations/auth0/edit.tsx [hunk 3]
```diff
@@ -1,30 +1,105 @@
...
-		>
+		<form className="mt-6 flex flex-col gap-6" onSubmit={openOauthPopUp}>
+			<div className="relative">
+				<Input
+					{...register("client_id")}
+					aria-label={t("auth0.placeholders.client_id")}
+					isError={!!errors.client_id}
+					isRequired
+					label={t("auth0.placeholders.client_id")}
+					value={clientId}
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/organisms/connections/integrations/auth0/edit.tsx:49]: Yes

**Review by J1za — APPROVED** [review]

**Pull request merged**